### PR TITLE
BUG: force list type for tuples from chunked sql table reads #11522

### LINKED ISSUE #11522: read_sql: TypeError: Argument 'rows' has incorrect type
Hi,
I encountered a problem when using read_sql with a remote mySQL database.

After loading the modules,
```python
import pymysql
import pandas as pd
import sqlalchemy
```

I create both a `pymysql` connector and a `sqlalchemy` engine with `pymysql` for the same database.
```python
connector = pymysql.connect(...)
engine = sqlalchemy.create_engine('mysql+pymysql://...')
```

Everything is fine with the engine
```python
df = pd.DataFrame()
query = " SELECT id, created_at FROM themes"
for chunk in pd.read_sql(query, engine, chunksize=100):
    df = df.append(chunk)
```
but fails with the connector
```python
df = pd.DataFrame()
query = " SELECT id, created_at FROM themes"
for chunk in pd.read_sql(query, connector, chunksize=100):
    df = df.append(chunk)

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-7-9a0ee2188a78> in <module>()
      1 df = pd.DataFrame()
      2 query = " SELECT id, created_at FROM themes"
----> 3 for chunk in pd.read_sql(query, connector_mySQL, chunksize=100):
      4     df = df.append(chunk)

/usr/local/lib/python3.5/site-packages/pandas/io/sql.py in _query_iterator(cursor, chunksize, columns, index_col, coerce_float, parse_dates)
   1563                 yield _wrap_result(data, columns, index_col=index_col,
   1564                                    coerce_float=coerce_float,
-> 1565                                    parse_dates=parse_dates)
   1566 
   1567     def read_query(self, sql, index_col=None, coerce_float=True, params=None,

/usr/local/lib/python3.5/site-packages/pandas/io/sql.py in _wrap_result(data, columns, index_col, coerce_float, parse_dates)
    135 
    136     frame = DataFrame.from_records(data, columns=columns,
--> 137                                    coerce_float=coerce_float)
    138 
    139     _parse_date_columns(frame, parse_dates)

/usr/local/lib/python3.5/site-packages/pandas/core/frame.py in from_records(cls, data, index, exclude, columns, coerce_float, nrows)
    928         else:
    929             arrays, arr_columns = _to_arrays(data, columns,
--> 930                                              coerce_float=coerce_float)
    931 
    932             arr_columns = _ensure_index(arr_columns)

/usr/local/lib/python3.5/site-packages/pandas/core/frame.py in _to_arrays(data, columns, coerce_float, dtype)
   5211     if isinstance(data[0], (list, tuple)):
   5212         return _list_to_arrays(data, columns, coerce_float=coerce_float,
-> 5213                                dtype=dtype)
   5214     elif isinstance(data[0], collections.Mapping):
   5215         return _list_of_dict_to_arrays(data, columns,

/usr/local/lib/python3.5/site-packages/pandas/core/frame.py in _list_to_arrays(data, columns, coerce_float, dtype)
   5289 def _list_to_arrays(data, columns, coerce_float=False, dtype=None):
   5290     if len(data) > 0 and isinstance(data[0], tuple):
-> 5291         content = list(lib.to_object_array_tuples(data).T)
   5292     else:
   5293         # list of lists

TypeError: Argument 'rows' has incorrect type (expected list, got tuple)
```

Here are the versions of the modules
```python
print(pymysql.__version__)
print(sqlalchemy.__version__)
print(pd.__version__)

0.6.7.None
1.0.9
0.17.0
```
Best regards,
Flavien.

### PR DESCRIPTION
**Pull request by grahamjeffries**
Closes #11522.

### CODE REVIEW

File: pandas/io/tests/test_sql.py [hunk 1]
```diff
@@ -2477,6 +2477,25 @@ def test_write_row_by_row(self):
...
+        frame.index.name = "index"
+        drop_sql = "DROP TABLE IF EXISTS test"
+        cur = self.conn.cursor()
+        cur.execute(drop_sql)
+        sql.to_sql(frame, name='test', con=self.conn, flavor='mysql')
+        query = "select * from test"
+        chunksize = 5
+        chunk_gen = pd.read_sql_query(sql=query, con=self.conn,
+                                      chunksize=chunksize, index_col="index")
+        chunk_df = pd.DataFrame()
```
Comment (jorisvandenbossche) [inline pandas/io/tests/test_sql.py]: This line is superfluous

File: pandas/io/tests/test_sql.py [hunk 2]
```diff
@@ -2477,6 +2477,25 @@ def test_write_row_by_row(self):
...
+        try:
+            chunk_df = next(chunk_gen)
+        except TypeError:
+            assert False, "expected list of tuples, got tuple of tuples"
```
Comment (jorisvandenbossche) [inline pandas/io/tests/test_sql.py]: I don't think the try .. except is necessary, the test will just error if this does not work, which is OK as well

**Comment by jorisvandenbossche** [comment]
@grahamjeffries Looks good! thanks

This is actually in code that is deprecated, and will be removed in the future. But as it is only a small change, ok to let this in for the case it will not yet be removed in the next version.

**Comment by grahamjeffries** [comment]
Sure thing. Yeah, I wasn't quite sure what the protocol was for testing cases which fail with an exception.

**Comment by jorisvandenbossche** [comment]
@grahamjeffries OK! Just a last question, can you add a note in the whatsnew file (doc/source/whatsnew/v0.18.0.txt) in the bug fixes section, and then also squash the commits? (http://pandas.pydata.org/pandas-docs/stable/contributing.html#combining-commits)

**Comment by grahamjeffries** [comment]
Noted and squashed.

**Comment by jorisvandenbossche** [comment]
Thanks a lot!

**Pull request merged**